src/colmap/exe/sfm.cc

```diff
@@ -392,6 +392,26 @@ int RunPosePriorMapper(int argc, char** argv) {
                            &options.mapper->use_robust_loss_on_prior_position);
   options.AddDefaultOption("prior_position_loss_scale",
                            &options.mapper->prior_position_loss_scale);
+
+  std::string cartesian_frame = "enu";
+  options.AddDefaultOption(
+      "cartesian_frame",
+      &cartesian_frame,
+      "Specifies the Cartesian coordinate frame for converting from latitude, "
+      "longitude, and altitude (LLA). Possible values are 'ENU', "
+      "'ECEF', and 'UTM'.");
+
+  StringToLower(&cartesian_frame);
+  if (cartesian_frame == "enu") {
```

Comment:
Use a switch statement instead.

Reply:
The condition in switch statement can only be integral/enumeration/class(implicitly converted to integral/enumeration) for [C++17](https://en.cppreference.com/w/cpp/language/switch), and colmap seems don't have a method to transform string to enum(only have enum to string). So, to use switch statement here

- Should we consider to add function like `StringToEnum`?
- Or should we consider to transform string to some unique integral(e.g. hash value)?

Reply:
I would keep it as is and address this separately in another PR, if we can make our macros produce a StringToEnum function. This can then also be applied to various other places with enums.

Reply:
Don't we already have this for enums defined with `MAKE_ENUM`?
https://github.com/colmap/colmap/blob/590c5091590e3ea97c3a87229a688f56b643ce62/src/colmap/geometry/gps.cc#L376

https://github.com/colmap/colmap/blob/590c5091590e3ea97c3a87229a688f56b643ce62/src/colmap/util/enum_to_string.h#L54-L55